test/tap/groups/check_groups.py

```
@@ -53,6 +63,37 @@ def find_executable_tests(tap_root):
+    the source tree, so it works on a fresh checkout in a lint CI job.
+    """
+    scan_roots = [
+        os.path.join(tap_root, "tests"),
+        os.path.join(tap_root, "tests_with_deps"),
+    ]
+
+    sources = set()
+    for scan_root in scan_roots:
+        if not os.path.isdir(scan_root):
+            continue
+        for dirpath, _, filenames in os.walk(scan_root):
```

Ruff (0.15.10)

[warning] Loop control variable `dirpath` not used within loop body

Rename unused `dirpath` to `_dirpath`

(B007)